diff --git a/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt b/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
index 7c387ab..c38e102 100644
--- a/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
+++ b/node_modules/expo-modules-core/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
@@ -14,6 +14,9 @@ import java.util.Properties
  * This can add up - it's better to run it once and cache the result.
  */
 open class ExpoGradleHelperExtension {
+  private val nodeExecutable = System.getenv("NODE_BINARY")
+    ?: "/Users/manojkumar/.nvm/versions/node/v20.11.1/bin/node"
+
   /**
    * Cached React Native directory
    */
@@ -43,7 +46,7 @@ open class ExpoGradleHelperExtension {
     reactNativeDir = reactNativeDirFromSource ?: File(
       project.providers.exec { env ->
         env.workingDir(project.rootDir)
-        env.commandLine("node", "--print", "require.resolve('react-native/package.json')")
+        env.commandLine(nodeExecutable, "--print", "require.resolve('react-native/package.json')")
       }.standardOutput.asText.get().trim()
     ).parentFile
 
