diff --git a/node_modules/react-native-screens/android/build.gradle b/node_modules/react-native-screens/android/build.gradle
index 114be81..92b493c 100644
--- a/node_modules/react-native-screens/android/build.gradle
+++ b/node_modules/react-native-screens/android/build.gradle
@@ -63,7 +63,7 @@ def resolveReactNativeDirectory() {
     // We're in non standard setup, e.g. monorepo - try to use node resolver to locate the react-native package.
     String maybeRnPackagePath = providers.exec {
         workingDir(rootDir)
-        commandLine("node", "--print", "require.resolve('react-native/package.json')")
+        commandLine(System.getenv("NODE_BINARY") ?: "/Users/manojkumar/.nvm/versions/node/v20.11.1/bin/node", "--print", "require.resolve('react-native/package.json')")
     }.standardOutput.asText.get().trim()
 
     File nodeResolverRnDirFile = null
